include/oneapi/dpl/pstl/experimental/internal/for_loop_impl.h

```diff
@@ -269,9 +269,9 @@ __execute_loop_strided(_Ip __first, _Ip __last, _Function __f, _Sp __stride, _Pa
 }
 
 template <typename _Ip, typename _Function, typename _Sp, typename _Pack, typename _IndexType>
-::std::enable_if_t<::std::is_same_v<typename oneapi::dpl::__internal::__iterator_traits<_Ip>::iterator_category,
+::std::enable_if_t<::std::is_same_v<typename ::std::iterator_traits<_Ip>::iterator_category,
```

Comment:
Do we whether want to use `is_base_of(input_iterator tag) ` here?
Because a 'forward' iterator is an 'input' iterator as well.

Reply:
I have checked the code and `__execute_loop_strided` only uses for bidirectional and forward iterators. I have changed the code a bit to use the newly introduced trait and removed `enable_if` here.